### PR TITLE
New header search bar

### PR DESCRIPTION
**Pull request by aintgoin2goa**

### CODE REVIEW

File: footer/partials/sidenav.html [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
-		<a class="o-header__sidenav-section-link" href="{{href}}" {{#if selected}}aria-selected="true"{{/if}} data-trackable="{{name}}">Go to {{name}}</a>
-		<a class="o-header__sidenav-section-link o-header__sidenav-section-link--nojs" href="{{href}}" {{#if selected}}aria-selected="true"{{/if}} data-trackable="{{name}}">{{name}}</a>
-		{{else}}
-		<a class="o-header__sidenav-section-title" href="{{href}}">{{name}}</a>
-		{{/if}}
+<div class="o-header__sidenav-drawer js-sidenav">
+	<form class="o-header__sidenav-search" action="/search" data-o-grid-colspan="12">
+		<label class="o-header__sidenav-search__label" for="o-header__sidenav-search__term">Type to search the Financial Times</label>
+		<div class="o-header__sidenav-search__input-wrapper">
+			<input id="o-header__sidenav-search__term" name="q" class="o-header__sidenav-search__term" type="search" data-trackable="search-term" placeholder="Type to search the FT" />
```
Comment (andygout) [inline footer/partials/sidenav.html]: Quite cryptic `name="q"` - what does this do?

File: header/_sidenav.scss [hunk 2]
```diff
@@ -37,23 +43,51 @@
...
 
-	&--visible {
-		display: block;
+	.o-header__sidenav-content-mask {
+		position: fixed;
+		top: 0;
+		left: 0;
+		bottom: 0;
+		right: 0;
+		background-color: #000000;
```
Comment (andygout) [inline header/_sidenav.scss]: Am guessing there is no 'official' colour variable we should be using for black here?

File: main.js [hunk 3]
```diff
@@ -2,42 +2,36 @@ const Typeahead = require('./header/typeahead');
...
+	}
+	sidenav.init(flags);
+	require('o-header').init();
+
+
+
+
+
+
+	const suggestionsContainer = document.querySelector('[data-o-header-suggestions]');
```
Comment (andygout) [inline main.js]: Lotta empty lines here.

File: footer/partials/sidenav.html (hunk 1 above)
Comment (aintgoin2goa) [inline footer/partials/sidenav.html]: Thats the search bar - I just copied it from existing one.  I presume it's cos the search page or search or whatever expects "?q=mysearch"

File: header/_sidenav.scss (hunk 2 above)
Comment (aintgoin2goa) [inline header/_sidenav.scss]: Yes, probably - but not got real colours yet so using this until I know

File: footer/partials/sidenav.html (hunk 1 above)
Comment (andygout) [inline footer/partials/sidenav.html]: Ah, right.

**Comment by andygout** [comment]
👍

**Pull request merged**